charts/k6-operator/templates/role.yaml

```
@@ -32,9 +70,193 @@ rules:
   - list
   - update
 - apiGroups:
-    - ""
+  - ""
+  resources:
+  - pods
+  - pods/log
+  verbs:
+  - get
+  - list
+  - watch
+- apiGroups:
+  - k6.io
+  resources:
+  - k6s
+  verbs:
+  - create
+  - delete
+  - get
+  - list
+  - patch
+  - update
+  - watch
+- apiGroups:
+  - k6.io
+  resources:
+  - k6s/finalizers
+  - k6s/status
+  verbs:
+  - get
+  - patch
+  - update
+- apiGroups:
+  - k6.io
+  resources:
+  - privateloadzones
+  verbs:
+  - create
+  - delete
+  - get
+  - list
+  - patch
+  - update
+  - watch
+- apiGroups:
+  - k6.io
+  resources:
+  - privateloadzones/finalizers
+  verbs:
+  - get
+  - patch
+  - update
+- apiGroups:
+  - k6.io
+  resources:
+  - privateloadzones/status
+  verbs:
+  - get
+  - patch
+  - update
+- apiGroups:
+  - k6.io
+  resources:
+  - testruns
+  verbs:
+  - create
+  - delete
+  - get
+  - list
+  - patch
+  - update
+  - watch
+- apiGroups:
+  - k6.io
+  resources:
+  - testruns/finalizers
+  - testruns/status
+  verbs:
+  - get
+  - patch
+  - update
+---
+apiVersion: rbac.authorization.k8s.io/v1
+{{- if not .Values.rbac.namespaced }}
+kind: ClusterRole
+{{- else }}
+kind: Role
+{{- end }}
+metadata:
+  name: {{ include "k6-operator.fullname" . }}-metrics-auth-role
+  {{- if .Values.rbac.namespaced }}
+  namespace: {{- include "k6-operator.namespace" . }}
+  {{- end }}
+  labels:
+    {{- include "k6-operator.labels" . | nindent 4 }}
+    {{- include "k6-operator.customLabels" . | default "" | nindent 4 }}
+  annotations:
+    {{- include "k6-operator.customAnnotations" . | default "" | nindent 4 }}
+rules:
+- apiGroups:
+  - authentication.k8s.io
   resources:
-    - events
+  - tokenreviews
+  verbs:
+  - create
+- apiGroups:
+  - authorization.k8s.io
+  resources:
+  - subjectaccessreviews
+  verbs:
+  - create
+---
+apiVersion: rbac.authorization.k8s.io/v1
+{{- if not .Values.rbac.namespaced }}
+kind: ClusterRole
+{{- else }}
+kind: Role
+{{- end }}
+metadata:
+  name: {{ include "k6-operator.fullname" . }}-metrics-reader
+  {{- if .Values.rbac.namespaced }}
+  namespace: {{- include "k6-operator.namespace" . }}
+  {{- end }}
+  labels:
+    {{- include "k6-operator.labels" . | nindent 4 }}
+    {{- include "k6-operator.customLabels" . | default "" | nindent 4 }}
+  annotations:
+    {{- include "k6-operator.customAnnotations" . | default "" | nindent 4 }}
+rules:
+- nonResourceURLs:
+  - /metrics
   verbs:
-    - create
-    - patch
\ No newline at end of file
+  - get
```

Error: INSTALLATION FAILED: 1 error occurred:
        * Role.rbac.authorization.k8s.io "k6-operator-metrics-reader" is invalid: rules[0].nonResourceURLs: Invalid value: ["/metrics"]: namespaced rules cannot apply to non-resource URLs

```suggestion
{{- if not .Values.rbac.namespaced }}
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: {{ include "k6-operator.fullname" . }}-metrics-reader
  {{- if .Values.rbac.namespaced }}
  namespace: {{- include "k6-operator.namespace" . }}
  {{- end }}
  labels:
    {{- include "k6-operator.labels" . | nindent 4 }}
    {{- include "k6-operator.customLabels" . | default "" | nindent 4 }}
  annotations:
    {{- include "k6-operator.customAnnotations" . | default "" | nindent 4 }}
rules:
- nonResourceURLs:
  - /metrics
  verbs:
  - get
{{- end }}
```